src/main/kotlin/view/fragment/TransformationList.kt

```diff
@@ -31,12 +31,11 @@ class TransformationList : Fragment() {
 
             listview(engine.transformations) {
                 prefWidth = 400.0
-                selectionModel.selectedIndexProperty()
-                    .onChange {
-                        engine.setCurrentIndex(it)
+                engine.currIndexProperty.addListener(
+                    ChangeListener { _, _, newValue ->
+                        selectionModel.select(newValue.toInt() - 1)
```

Comment:
`newValue - 1` is the hacky bit since the index in EngineModel is always 1 unit too big. Check that it works all the time please!